test/correctness/simd_op_check.cpp

```diff
@@ -1871,6 +1871,16 @@ struct Test {
         check("vmpa(v*.h,r*.b)", hvx_width/2, 2*i32(i16_1) + 3*i32(i16_2));
         check("v*.w += vmpa(v*.h,r*.b)", hvx_width/2, 2*i32(i16_1) + 3*i32(i16_2) + i32_1);
 
+        check("v*:*.h = vtmpy(v*:*.ub, r*.b)", hvx_width/1, 2*i16(in_u8(x + hvx_width - 1)) + 3*i16(in_u8(x + hvx_width)) + i16(in_u8(x + hvx_width + 1)));
```

Comment:
This can be fixed by modifying https://github.com/halide/Halide/blob/master/apps/simd_op_check/driver.cpp#L36 to allocate padding before the min, i.e. add 128 to w prior to allocating, and set dim[0].min to -128.